glslc/test/working_directory.py

```diff
@@ -207,11 +157,63 @@ class TestWorkDirCompileFileAbsolutePath(expect.ValidObjectFile):
     glslc_args = ['-c', '-working-directory=subdir', shader]
 
 
-@inside_glslc_testsuite('WorkDir')
-class TestWorkDirLinkFileAbsolutePath(expect.ValidNamedObjectFile):
-    """Tests -working-directory=<dir> when linking input file with absolute
-    path."""
+# The -working-directory flag should not affect the placement of the link file.
+# The following tests ensure that.
 
-    shader = FileShader('void main() {}', '.vert')
-    glslc_args = ['-working-directory=subdir', shader]
+class WorkDirDoesntAffectLinkedFile(expect.ValidNamedObjectFile):
+    """A base class for tests asserting that -workind-directory has no impact
+    on the location of the output link file.
+    """
+    environment = Directory('.', [
+        Directory('subdir', [
+            File('shader.vert', 'void main(){}'),
+            # This bin/ is in the wrong location, so glslc should ignore it.
+            Directory('bin', [])]),
+        File('shader.vert', "fake file, doesn't compile."),
+        Directory('bin', [])])
+
+
+@inside_glslc_testsuite('WorkDir')
+class TestWorkDirLinkFileDefaultLocation(WorkDirDoesntAffectLinkedFile):
+    """Tests that -working-directory doesn't impact the default link-file
+    location.
+    """
+    glslc_args = ['-working-directory=subdir', 'shader.vert']
     expected_object_filenames = ('a.spv',)
```

Comment:
Btw, the rules I used (which is from my experience with clang):
https://github.com/google/shaderc/tree/master/glslc#413--working-directory